// @fireworks-js/react/dist/index.es.js
/**
 * name: @fireworks-js/react
 * version: 2.2.0
 * author: Vitalij Ryndin (https://crashmax.ru)
 * homepage: https://fireworks.js.org
 * license MIT
 */
import { Fireworks as i } from "fireworks-js";
import n, { useRef as u, useImperativeHandle as p, useEffect as f } from "react";
const m = n.forwardRef(
  ({ children: s, options: a, ...c }, o) => {
    const t = u(null), r = u(null);
    return p(o, () => ({
      get isRunning() {
        return r.current.isRunning;
      },
      start() {
        r.current.start();
      },
      stop() {
        r.current.stop();
      },
      pause() {
        r.current.pause();
      },
      clear() {
        r.current.clear();
      },
      updateOptions(e) {
        r.current.updateOptions(e);
      },
      updateSize(e) {
        r.current.updateSize(e);
      }
    })), f(() => (r.current = new i(t.current, a), r.current.start(), () => {
      r.current.stop(!0);
    }), []), /* @__PURE__ */ n.createElement("div", {
      ref: t,
      ...c
    }, s);
  }
);
export {
  m as Fireworks,
  m as default
};
